vncviewer/DesktopWindow.cxx

```diff
@@ -465,9 +470,9 @@ void DesktopWindow::draw()
 
     // Make sure it's properly seen by adjusting it relative to the
     // primary screen rather than the entire window
-    if (fullscreen_active() && fullScreenAllMonitors) {
+    if (fullscreen_active() && (!strcmp(fullScreenMode, "Selected") || !strcmp(fullScreenMode, "All") || fullScreenAllMonitors)) {
```

Comment:
This is getting a bit long and difficult to read. Can we split things up better?